src/ACMESharp/Protocol/Messages/RevokeCertificateRequest.cs

```diff
@@ -0,0 +1,22 @@
+using System.ComponentModel.DataAnnotations;
+using Newtonsoft.Json;
+
+namespace ACMESharp.Protocol.Messages
+{
+    /// <summary>
+    /// https://tools.ietf.org/html/draft-ietf-acme-acme-18#section-7.6
+    /// </summary>
+    public class RevokeCertificateRequest
+    {
+        [JsonProperty("certificate", Required = Required.Always)]
+        [Required]
+        public string Certificate { get; set; }
+
+        // Possible reasons specified here
+        // https://tools.ietf.org/html/rfc5280#section-5.3.1
+        // Not sure where best to create an enum and how to handle the (optional) 
+        // serialization of this property, so leaving it out for now
+        //[JsonProperty("reason")]
+        //public int Reason { get; set; }
```

Comment:
Since this is actually specified as an int in the spec, I think we can safely use a standard enum with explicit number mapping as per the second spec.  Serializing enums as integers is the default behavior for Json.NET, so that should be fine for the protocol exchange, but the enum will provide a better developer experience.  I can try to throw something together shortly after the first merge builds successfully.

Reply:
True. I was kind of mistaken here in thinking that it should be an optional parameter, but actually the list of reasons specifies a sensible default 0 ("unspecified") so it can be added into the message regardless of the caller specifying anything.